metaserver/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -435,6 +414,18 @@ object ScalametaLanguageServer extends LazyLogging {
     subscriber -> semanticdbPublisher
   }
 
+  def configurationStream(connection: Connection)(
+    implicit scheduler: Scheduler
+  ): (Observer.Sync[JsValue], Observable[Configuration]) = {
+    val (subscriber, publisher) = multicast[JsValue]
+    val configurationPublisher = publisher
+      .map(json => (json \ "scalameta").as[Configuration])
+      .doOnNext(conf => logger.info(s"Configuration updated $conf"))
+      .doOnError(e => connection.showMessage(MessageType.Error, e.getMessage))
+    subscriber.onNext(Json.toJson(Configuration()))
```

Comment:
this is where I bootstrap the configuration, to make sure the observable is never empty. Not super elegant, but it works.